src/modules/emote_autocomplete/twitch/EmoteAutocomplete.jsx

```diff
@@ -0,0 +1,169 @@
+import twitch from '../../../utils/twitch.js';
+import watcher from '../../../watcher.js';
+import emoteMenuViewStore from '../../../common/stores/emote-menu-view-store.js';
+import dom from '../../../observers/dom.js';
+import emotes from '../../emotes/index.js';
+import {createSrcSet} from '../../../utils/image.js';
+import {SettingIds} from '../../../constants.js';
+import settings from '../../../settings.js';
+import './styles.module.css';
+
+const EMOTE_SET_CODE_PREFIX = '__bttv-';
+const AUTOCOMPLETE_MATCH_IMAGE_QUERY = '.emote-autocomplete-provider__image, .chat-line__message--emote';
+
+function seralizeCode(code) {
+  return `${EMOTE_SET_CODE_PREFIX}${code}`;
```

Comment:
it may be a bit safer to serialize and deserialize using some url-safe encoding

```
function serializeEmoteId(emote) {
    const data = `${emote.category.provider}-${emote.id}-${emote.code}`;
    return `${EMOTE_ID_BETTERTTV_PREFIX}${encodeURIComponent(btoa(data))}`;
}

function deserializeEmoteFromURL(url) {
    const emoteData = emoteId.split(EMOTE_ID_BETTERTTV_PREFIX)[1]?.split('/')[0];
    if (emoteData == null) {
        return null;
    }
    try {
        const [emoteProvider, emoteId, emoteCode] = atob(decodeURIComponent(emoteData)).split('-');
        return {
            provider: emoteProvider,
            id: emoteId,
            code: emoteCode,
        };
    } catch(_) {
        return null;
    }
}
```